hs_core/views/__init__.py

```diff
@@ -69,6 +72,38 @@ def add_file_to_resource(request, shortkey, *args, **kwargs):
     extract_metadata = request.REQUEST.get('extract-metadata', 'No')
     extract_metadata = True if extract_metadata.lower() == 'yes' else False
 
+    irods_fname = request.session.get('irods_add_file_name', '')
+    if irods_fname:
+        res_cls = resource.__class__
+        file_types = res_cls.get_supported_upload_file_types()
```

Comment:
if it is a tuple why do you do `file_types == ".*"` assuming it to be a string below?